### PR TITLE
API-16147 F: V1: Remove custom deserialization from API-13372 baseline

### PR DESCRIPTION
**Pull request by mmccollum-va**
Enumerated list of enum values updated to list of service summary objects for V1/Datamart Facility object services list.

### CODE REVIEW

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/ExceptionsUtils.java [hunk 1]
```diff
@@ -11,6 +11,13 @@ static final class InvalidParameter extends RuntimeException {
   }
 
   static final class NotFound extends RuntimeException {
+    public NotFound(String serviceId, String facilityId) {
+      super(
+          String.format(
+              "The overlay service identified by %s could not be found for facility %s",
```
Comment (as-va) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/ExceptionsUtils.java:17]: Let's take out the word "overlay" here. This is an external facing message and most consumers won't be familiar with the "overlay" concept - just that there are facilities and services.

File: facilities-api/src/main/java/gov/va/api/lighthouse/facilities/api/v1/Facility.java [hunk 2]
```diff
@@ -1018,23 +1026,31 @@ public static final class Services implements CanBeEmpty {
...
             @Schema(
-                description = "List of benefits services " + "for given facility.",
+                description = "List of benefits service objects " + "for given facility.",
                 nullable = true))
-    List<BenefitsService> benefits;
+    List<Service<BenefitsService>> benefits;
+
+    @Schema(
+        description = "Base services link for services at facility.",
+        example = "http://localhost/v1/facilities/vha_558GA/services/",
```
[the remark this answers is not in the document]
Comment (as-va) [reply facilities-api/src/main/java/gov/va/api/lighthouse/facilities/api/v1/Facility.java:1018]: Also applies to the individual service example below.

[the remark this answers is not in the document]
Comment (mmccollum-va) [reply facilities-api/src/main/java/gov/va/api/lighthouse/facilities/api/v1/Facility.java]: This has been addressed

File: facilities-api/src/main/java/gov/va/api/lighthouse/facilities/api/v1/Facility.java [hunk 3]
```diff
@@ -971,6 +984,7 @@ public boolean isEmpty() {
   @Builder
   @JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
   @JsonInclude(value = Include.NON_EMPTY, content = Include.NON_EMPTY)
+  @JsonPropertyOrder({"other", "health", "benefits", "link", "lastUpdated"})
```
[the remark this answers is not in the document]
Comment (mmccollum-va) [reply facilities-api/src/main/java/gov/va/api/lighthouse/facilities/api/v1/Facility.java]: This has been addressed

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/ExceptionsUtils.java (hunk 1 above)
Comment (mmccollum-va) [reply facilities/src/main/java/gov/va/api/lighthouse/facilities/ExceptionsUtils.java]: This has been addressed

**Comment by JamesDanielBell-VA** [comment]
If we post a CMS payload where a service name is not in the HealthService enum(like smoking), do we expect to return lower case or upper case in response?
I notice that Covid19 is upper case ,but smoking is lower.

response
payload

**Comment by mmccollum-va** [comment]
This is an artifact of allowing V1 facility services to be uploaded through the V0 endpoint:
- Using the payload posted above, the V0 upload endpoint does recognize `smoking` as a valid service, but since it does match any of the V0 facility health services, it will honor the `smoking` name provided in the payload.
- Uploading using the V1 endpoint, we see the following overlay output for facility:
```
{
    "overlay": {
        "detailedServices": [
            {
                "serviceInfo": {
                    "name": "Covid19Vaccine",
                    "serviceId": "covid19Vaccine",
                    "serviceType": "health"
                },
                "appointmentLeadIn": "Your VA health care team will contact you if you...more text",
                "onlineSchedulingAvailable": "Unknown",
                "referralRequired": "false",
                "walkInsAccepted": "true"
            },
            {
                "serviceInfo": {
                    "name": "Smoking",
                    "serviceId": "smoking",
                    "serviceType": "health"
                },
                "appointmentLeadIn": "Your VA health care team will contact you if you...more text",
                "onlineSchedulingAvailable": "Unknown",
                "referralRequired": "false",
                "walkInsAccepted": "true"
            }
        ]
    }
}
```

**Review by JamesDanielBell-VA — APPROVED** [review]

**Pull request merged**